### ISSUE TITLE
.eslintignore does not work with symlinked dependencies

### ISSUE THREAD

**Issue by jrencz** [MoOx/eslint-loader#202]
I observed strange behaviour in my setup and I'm not 100% sure if it's `eslint-loader`, but maybe it is.

I have my applications composed from public packages and from several in-house packages published to private NPM instance.

Those in-house dependencies are most of the time `npm link`ed in applications which allows seamless development.

The problem is that in case I modify installed dependency (the one that is not a result of executing `npm link <dependency package name>`) it's not linted
I fI modify a linked dependecy is changed then it's linted

Both main project (where webpack runs) and the linked dependency have `.eslintignore` files:

```
/node_modules/*
/dist/*
```

Main project has loader config (webpack1):

```js
{
      test: /\.js$/,
      loader: 'eslint-loader',
}
```
(I expect the loader not to have to have `exclude` if there is`.eslintignore`)

Steps to reproduce:
1) create webpack project with at least 2 dependencies and `eslint-loader`. add some rules which are not violated in project code but are violated in dependency code (easiest: make sure dependencies are minified. It's going to violate virtually any rule)
2) prepare a working copy of one of dependencies, make sure both main project and dependency working copy use the same version of node in order to be able to link.
3) execute `npm link` in the working copy of one of dependencies. 
4) start webpack in the main project
5) go to non-linked dependency in main project `node_modules` and modify source
6) observe that project was rebuilt and there are no eslint errors
7) go to linked dependency working copy and modify the source
8) observe that project was rebuilt and there are some eslint errors

**Comment by MoOx** [comment]
The best way to confirm a bug, and have it fixed quickly is by producing a PR with a failing test. I won't have time to try to reproduce this very specific use case.

**Comment by jrencz** [comment]
I gave it a try by copying the existing `eslintignore.js` test and trying to symlink the `./test/fixtures/ignore.js` file but the test passes.
I couldn't create link inside node_modules because tests config ignores node_modules (and I don't think undoing that would change anything in terms of that particular test)

I investigated the problem a bit more by trying to run ESLint CLI agains 2 files: one from linked dependency and one from installed: the one from installed triggered `  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to override` warning while the one from linked did not.

It turns out ESLint resolves paths before matching against `.eslintignore` rules so the fact that my linked dependency is linked means nothing to ESLint: it won't match against `node_modules`.

In my case turning `dist/*` to `**/dist/*` did the trick.
